notes/12-guarding-routes.md

```diff
@@ -0,0 +1,195 @@
+# Guarding Routes
+
+In this exercise, we will be adding redirection in the UI flow based on the availability of an authenticated session.
+Specifically, users should be able to access the `login` route, only when an authenticated session _does not exist_. And similarly, users should be able to access the `teams` route only when an authenticated _session exists_.
+
+So let's get started.
+
+## Changes Needed in Javascript
+
+First let's implement the auth service that we will be using to do session validation(or invalidation). This auth service is defined at `app/services/auth.js`.
+
+Import the action decorator at the top of the file:
+
+```js
+import { action } from '@ember/object';
+```
+
+Then add a property, `isAuthenticated` that will be used to store if the session is in an authenticated state, and an action method, `logout`, for invalidating the authenticated session and signing off the user.
```

Comment:
```suggestion
Then add a getter-based property, `isAuthenticated` that will be `true` if the user is logged in and `false` otherwise. Add an action called `logout` that clears the user ID we write to localStorage on login, and sends the user to the `/login` page.
```
Simplifying some of the language. I'm hopeful we can avoid talking about a "session" and just talk in terms of an ID stored somewhere.